channeldb/forwarding_log.go

```diff
@@ -272,3 +306,47 @@ func (f *ForwardingLog) Query(q ForwardingEventQuery) (ForwardingLogTimeSlice, e
 
 	return resp, nil
 }
+
+// makeUniqueTimestamps takes a slice of forwarding events, sorts it by the
+// event timestamps and then makes sure there are no duplicates in the
+// timestamps. If duplicates are found, some of the timestamps are increased on
+// the nanosecond scale until only unique values remain. This is a fix to
+// address the problem that in some environments (looking at you, Windows) the
+// system clock has such a bad resolution that two serial invocations of
+// time.Now() might return the same timestamp, even if some time has elapsed
+// between the calls.
+func makeUniqueTimestamps(events []ForwardingEvent) {
+	sort.Slice(events, func(i, j int) bool {
+		return events[i].Timestamp.Before(events[j].Timestamp)
+	})
+
+	// Now that we know the events are sorted by timestamp, we can go
+	// through the list and fix all duplicates until only unique values
+	// remain.
+	for outer := 0; outer < len(events)-1; outer++ {
+		current := events[outer].Timestamp.UnixNano()
+		next := events[outer+1].Timestamp.UnixNano()
+
+		// If the current entry and the next are different, we can look
+		// at the next pair.
+		if current != next {
+			continue
+		}
+
+		// If the next entry is equal to the current, we go through the
+		// rest of the list and increase all timestamps that are equal
+		// to the current one, starting by the "next" one. This will
+		// shift all duplicates forward and over time fill "gaps"
+		// between the values, while always keeping the timestamps
+		// sorted in an incrementing manner.
+		for inner := outer + 1; inner < len(events); inner++ {
```

Comment:
isn't it enough to just check whether `current >= next`, set the timestamp of `next = current+1`, then go on to the next pair? Think it would simplify the code  a bit, and also avoid the nested loops.

Reply:
Ah, wow. Much simpler indeed, thank you!